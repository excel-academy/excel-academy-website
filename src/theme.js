import { themeColor } from '../site-config';

const purples = ['#7b79b0', '#6f6da5', '#eeeefc', '#1c2070'];
const greens = ['#d2eced', '#132c2d'];
const fontSizes = [12, 14, 16, 18, 20, 24, 30, 38, 62];
const space = [0, '2rem', '3rem', '4rem', '5rem', '7rem', '32rem', '40rem', '48rem', '56rem', '64rem'];
export const breakpoints = {
  sm: 0,
  mobile: '30em',
  tablet: '46em',
  desktop: '61.25em',
  wide: '80em',
  extrawide: '105em',
};

export default {
  space,
  fontSizes,
  breakpoints,
  maxWidths: ['48em', '60em', '80em'],
  colors: {
    theme: themeColor,
    purples,
    greens,
  },
  buttons: {
    primary: {
      color: '#fff',
      backgroundColor: themeColor,
      fontSize: fontSizes[4],
      borderColor: '#fff',
      paddingRight: space[4],
      paddingLeft: space[4],
      '&:hover': {
        backgroundColor: '#000',
      },
    },
    outline: {
      color: themeColor,
      backgroundColor: 'transparent',
      boxShadow: 'inset 0 0 2px',
    },
  },
};
