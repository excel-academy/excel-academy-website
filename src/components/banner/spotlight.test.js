import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from '../../theme';
import SpotlightBanner from './spotlight';
import image from '../../utils/mockBannerImg';

describe('SpotlightBanner', () => {
  it('left orient renders correctly', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <SpotlightBanner image={image.src} imagedesc={image.alt} contentAlign="left" orient="left" imagePosition="right">
          <h2>Title</h2>
          <p>Description</p>
        </SpotlightBanner>
      </ThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('right orient renders correctly', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <SpotlightBanner image={image.src} imagedesc={image.alt} contentAlign="left" orient="right" imagePosition="left">
          <h2>Title</h2>
          <p>Description</p>
        </SpotlightBanner>
      </ThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
