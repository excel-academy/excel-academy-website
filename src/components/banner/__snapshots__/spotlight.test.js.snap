// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightBanner left orient renders correctly 1`] = `
.c3 {
  box-sizing: border-box;
  padding: 2rem;
  width: 50%;
  z-index: 1;
}

.c5 {
  box-sizing: border-box;
  width: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5 .gatsby-image-wrapper {
  height: 100%;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: hidden;
  text-align: left;
}

.c1 .c4 img {
  object-position: right !important;
}

.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0 .c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0 auto;
}

@media screen and (min-width:61.25em) {
  .c3 {
    width: 66.66666666666666%;
  }
}

@media screen and (min-width:61.25em) {
  .c5 {
    width: 33.33333333333333%;
  }
}

<section
  className="c0 c1"
  orient="left"
>
  <div
    className="c2 c3"
    width={
      Object {
        "_": 0.5,
        "desktop": 0.6666666666666666,
      }
    }
  >
    <h2>
      Title
    </h2>
    <p>
      Description
    </p>
  </div>
  <div
    className="c4 c5"
    width={
      Object {
        "_": 0.5,
        "desktop": 0.3333333333333333,
      }
    }
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": "66.66666666666667%",
            "width": "100%",
          }
        }
      />
      <picture>
        <img
          alt="banner"
          loading="lazy"
          onLoad={[Function]}
          sizes="(max-width: 600px) 100vw, 600px"
          src="test-file-stub"
          srcSet="some srcSet"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><img loading=\\"lazy\\" sizes=\\"(max-width: 600px) 100vw, 600px\\" srcset=\\"some srcSet\\" src=\\"test-file-stub\\" alt=\\"banner\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
</section>
`;

exports[`SpotlightBanner right orient renders correctly 1`] = `
.c3 {
  box-sizing: border-box;
  padding: 2rem;
  width: 50%;
  z-index: 1;
}

.c5 {
  box-sizing: border-box;
  width: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5 .gatsby-image-wrapper {
  height: 100%;
}

.c6 .c4 img {
  object-position: right !important;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: hidden;
  text-align: left;
}

.c1 .c4 img {
  object-position: left !important;
}

.c7 .c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0 auto;
}

.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c0 .c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0 auto;
}

@media screen and (min-width:61.25em) {
  .c3 {
    width: 66.66666666666666%;
  }
}

@media screen and (min-width:61.25em) {
  .c5 {
    width: 33.33333333333333%;
  }
}

<section
  className="c0 c1"
  orient="right"
>
  <div
    className="c2 c3"
    width={
      Object {
        "_": 0.5,
        "desktop": 0.6666666666666666,
      }
    }
  >
    <h2>
      Title
    </h2>
    <p>
      Description
    </p>
  </div>
  <div
    className="c4 c5"
    width={
      Object {
        "_": 0.5,
        "desktop": 0.3333333333333333,
      }
    }
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": "66.66666666666667%",
            "width": "100%",
          }
        }
      />
      <picture>
        <img
          alt="banner"
          loading="lazy"
          onLoad={[Function]}
          sizes="(max-width: 600px) 100vw, 600px"
          src="test-file-stub"
          srcSet="some srcSet"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><img loading=\\"lazy\\" sizes=\\"(max-width: 600px) 100vw, 600px\\" srcset=\\"some srcSet\\" src=\\"test-file-stub\\" alt=\\"banner\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
</section>
`;
